pkg/filesystem/local_directory_windows.go

```diff
@@ -0,0 +1,1105 @@
+// +build windows
+
+package filesystem
+
+import (
+	"io"
+	"os"
+	"path/filepath"
+	"runtime"
+	"sort"
+	"strings"
+	"syscall"
+	"time"
+	"unsafe"
+
+	"github.com/buildbarn/bb-storage/pkg/filesystem/path"
+
+	"golang.org/x/sys/windows"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+var (
+	modntdll = windows.NewLazySystemDLL("ntdll.dll")
+
+	procNtSetInformationFile = modntdll.NewProc("NtSetInformationFile")
+)
+
+func ntSetInformationFile(handle windows.Handle, iosb *windows.IO_STATUS_BLOCK, fileInfo *byte, len uint32, class uint32) error {
+	r0, _, _ := syscall.Syscall6(procNtSetInformationFile.Addr(), 5, uintptr(handle),
+		uintptr(unsafe.Pointer(iosb)), uintptr(unsafe.Pointer(fileInfo)), uintptr(len), uintptr(class), 0)
+	if r0 != 0 {
+		return windows.NTStatus(r0)
+	}
+	return nil
+}
+
+const (
+	// these constants are not present in x/sys/windows
```

Comment:
I'm fine with keeping stuff like this in the tree in the short term, but could you please also file a PR against golang.org/x/sys to get this stuff added and add a comment to link to it? Thanks!

Reply:
I will start with creating PRs for adding the constants. This should be quick.

Adding NtSetFileInformation and the structs will likely need raising an issue and see what go dev thinks. I will do that as well.